src/mparticle-instance-manager.ts

```diff
@@ -17,21 +21,26 @@ if (!Array.prototype.filter) {
     Array.prototype.filter = Polyfill.filter;
 }
 
+// https://go.mparticle.com/work/SQDSDKS-6768
 if (!Array.isArray) {
+    // @ts-ignore
     Array.prototype.isArray = Polyfill.isArray;
 }
 
-function mParticle() {
-    var self = this;
+// Question: Should we just change this to mParticleInsanceManager?
+function mParticle(this: IMParticleInstanceManager, instanceName?: string) {
+    const self = this;
     // Only leaving this here in case any clients are trying to access mParticle.Store, to prevent from throwing
-    this.Store = {};
-    this._instances = {};
+    this.Store = {} as IStore;
+    this._instances = {} as Dictionary<IMParticleWebSDKInstance>;
     this.IdentityType = Types.IdentityType;
-    this.EventType = Types.EventType;
-    this.CommerceEventType = Types.CommerceEventType;
-    this.PromotionType = Types.PromotionActionType;
-    this.ProductActionType = Types.ProductActionType;
-    this.MPSideloadedKit = MPSideloadedKit;
+    this.EventType = EventType as unknown as valueof<typeof EventType>;
+    this.CommerceEventType = CommerceEventType as unknown as valueof<typeof CommerceEventType>;
+    this.PromotionType = PromotionActionType as unknown as valueof<typeof PromotionActionType>;
+    this.ProductActionType = ProductActionType as unknown as valueof<typeof ProductActionType>;
+
+    // QUESTION:  Should this be a 'new' instance?
+    this.MPSideloadedKit =  MPSideloadedKit as unknown as valueof<typeof MPSideloadedKit>;
```

Comment:
can you make a jira for this? i'd have to spend a bit of time (that i don't have right now) sorting through this one.

Reply:
Nope.  I also don't think it should be a valueof and just remain MPSideloadedKit

take a look at these docs in terms of how a customer uses it:
https://docs.mparticle.com/developers/client-sdks/web/kits/#including-the-sideloaded-kit